utils/tagrecommendation_utilities.py

```diff
@@ -78,10 +73,12 @@ def get_recommended_tags_view(request):
                 try:
                     tags, community = get_recommended_tags(input_tags)
                     return HttpResponse(json.dumps([tags, community]), content_type='application/javascript')
-                except Exception:
+                except urllib2.URLError, e:
+                    logger.error('Could not get a response from the tagrecommendation service (%s)\n\t%s' % \
+                         (e, traceback.format_exc()))
                     return HttpResponseUnavailabileError()
 
-    return HttpResponse(json.dumps([[],"-"]), content_type='application/javascript')
+    return HttpResponseUnavailabileError()
```

Comment:
I'm not sure this response is correct in this case - the returned data is "empty", which isn't an error case, it's just the case which is encountered when there is no data to respond to.
Looking at the javascript which processes this endpoint, it looks like there is a case which checks empty results: https://github.com/MTG/freesound/blob/132c96be9895ba67cd4a094882b36dec15876b95/templates/accounts/describe_sounds.html#L183
If this endpoint now returns 503, the js would have to be changed to perform this action on the POST failure. Instead, I think we should only return http503 in the single situation where the recommendation service is unavailable.